include/vrv/scoringupfunctor.h

```diff
@@ -0,0 +1,100 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        scoringupfunctor.h
+// Author:      Martha E. M. Thomae Elias
+// Created:     2024
+// Copyright (c) Authors and others. All rights reserved.
+/////////////////////////////////////////////////////////////////////////////
+///
+#ifndef __VRV_SCORINGUPFUNCTOR_H__
+#define __VRV_SCORINGUPFUNCTOR_H__
+
+#include "functor.h"
+
+namespace vrv {
+
+//----------------------------------------------------------------------------
+// ScoringUpFunctor
+//----------------------------------------------------------------------------
+
+/**
+ * This class...
+ */
+class ScoringUpFunctor : public Functor {
+public:
+    /**
+     * @name Constructors, destructors
+     */
+    ///@{
+    ScoringUpFunctor();
+    virtual ~ScoringUpFunctor() = default;
+    ///@}
+
+    /*
+     * Abstract base implementation
+     */
+    bool ImplementsEndInterface() const override { return true; }
+
+    /*
+     * Methods
+     */
+    /**
+     * @name: Divide the notes of a voice into sequences to be processed individualy
+     */
+    ///@{
+    std::vector<std::vector<std::pair<LayerElement *, data_DURATION>>> SubdivideSeq(
```

Comment:
You can add a `typedef` in `vrvdef.h` for the sequence (see https://github.com/rism-digital/verovio/blob/1ba7b68e7e398a2256feb39d42be9ebe2c4e1653/include/vrv/vrvdef.h#L336-L398)

e.g.,:
```cpp
typedef std::vector<std::pair<int, int>> ArrayOfElementDurPairs;
```

Then you would do:
```cpp
std::vector<ArrayOfElementDurPairs>
```

Reply:
Looking at this: since you are doing lots of inserts, you might want to use `std::list` instead of `std::vector`. You can also use `push_back` to insert at the end.

Reply:
> Looking at this: since you are doing lots of inserts, you might want to use `std::list` instead of `std::vector`. You can also use `push_back` to insert at the end.

@lpugin: You are right on this. I tried changing all vectors to lists and use `push_back` rather than `insert`, and everything seems ok with this. However, it is harder to use a list when I need to access a particular element in the list (right now I am using `.at()` for this) or when I need to extract a sublist (like I do for the `middleSeq` [here](https://github.com/martha-thomae/verovio/blob/bb92458c5bc058fc0c77ba1307c691759a8a2840/src/scoringupfunctor.cpp#L144)), unless I am missing something. I guess the biggest issue is when iterate through the elements of the `middleSeq` in the for loops to get each of these notes' values in minims:

https://github.com/martha-thomae/verovio/blob/bb92458c5bc058fc0c77ba1307c691759a8a2840/src/scoringupfunctor.cpp#L175-L186

Do you have any suggestions about this part? If you do, I can convert all these vectors into lists.

Reply:
If you need index access you should stick with `vector`. Don't worry about the performance issue for this.